web_widget_char_switchcase/static/src/js/web_widget_char_switchcase.js

```diff
@@ -41,13 +41,18 @@ odoo.define('web_widget_char_switchcase', function (require) {
             switch (this.current_transformation) {
                 case 'upper':
                     // HELLO WORLD!
-                    return function (val) { return val.toUpperCase(); };
+                    return function (val) {
```

Comment:
@SimoRubi In my example in former PR I've put this before the call to the transformer, exactly for this reason -> do not run the transform if there's nothing to transform.... I'd move this check before the call and leave this func as it is

Reply:
if you want to keep it here: move it before the switch

Reply:
@simahawk the check is there, inside the *keyup_handler*. Where would you put that too? I the `init` function?

Reply:
@simahawk I don't know if in the `init` I already have a value in the input field, actually I think it is not there yet, that's why I didn't check it in there.
In the `initialize_content` I might have the value already (after the `super` call), but the transformation_handler is already assigned and I think that's correct because the transformation_handler should not depend on the content but only on the option.
I could put the check in the `parse_value` and `format_value`, but I thought that checking the value can be something that the transformation_handler should manage by itself.

Then I've run out of functions! That's why the only solution I came up with is putting the check in each transformation_handler.
Where do you suggest to put the check?

Reply:
then how don't get how the module is supposed to work properly IF [this keyup event](https://github.com/OCA/web/blob/50e7cf588ea6bf8676dd0a0fc0b644f54d52477c/web_widget_char_switchcase/static/src/js/web_widget_char_switchcase.js#L24) is not triggered EVERY time you type.
If this is triggered properly then the handle should never be called. :thinking: 

Ahhhhh now I see: you are binding the event only if you have a value at initialize content (and we did not catch that in reviews :cry:).
I think you should delegate the binding to `events` mapping. Ideally I'd say you should add your own event to internal events (by extension) but it seems [there's already one that should do what we need](https://github.com/odoo/odoo/blob/1868713dbd07e0b518f91dffe73e62d85e6ab9a6/addons/web/static/src/js/fields/basic_fields.js#L157).
As you see, it says:
```js
/**
     * Called when the user is typing text -> By default this only calls a
     * debounced method to notify the outside world of the changes.
     * @see _doDebouncedAction
     *
     * @private
     */
    _onInput: function () {
        this.isDirty = true;
        this._doDebouncedAction();
    },
```
Hence, I'd say you should override it, do your stuff (what you do now inside the keyup callback) and call super.

Reply:
dammit, this would work in v11 (I was grepping in the wrong branch :stuck_out_tongue:). Then, the solution is:
```js
    events: _.extend({}, form_widgets.FieldChar.prototype.events, {
        'keyup': '_onKeyUp',
    },
    [...]
    _onKeyUp: function () {  ...your callback code here... }
```
This should do the trick.

Reply:
@simahawk used the new event binding in https://github.com/OCA/web/pull/1079/commits/9602babdb9809b0d68f2dc9877d7ea04e15079fe